pipeline_dp/private_beam.py

```diff
@@ -308,3 +315,117 @@ def __init__(self, fn: Callable, label: Optional[str] = None):
     def expand(self, pcol: pvalue.PCollection):
         return pcol | "flatten values" >> beam.ParDo(
             FlatMap._FlattenValues(map_fn=self._fn))
+
+
+class PrivateCombineFn(beam.CombineFn):
```

Comment:
IIUC, the developers implementing this method will need to understand differential privacy quite a lot, and will need to do things like clamping and noising. Maybe we can add some documentation to make it clear that this is an advanced feature that can break differential privacy guarantees if not implemented correctly, wdyt? 

Related: what is the distribution of responsibilities between the custom CombineFn and the rest of PipelineDp? I see that the custom code needs to take care of clamping and noising. What are the guarantees enforced by the PipelineDP, does it do cross-partition contribution bounding and partition selection?

(Also a few related questions below)